modules/raft/src/testFixtures/java/org/apache/ignite/raft/server/counter/IncrementAndGetCommand.java

```diff
@@ -18,29 +18,22 @@
 package org.apache.ignite.raft.server.counter;
 
 import org.apache.ignite.internal.raft.WriteCommand;
+import org.apache.ignite.network.annotations.Transferable;
+import org.apache.ignite.raft.messages.TestMessageGroup;
+import org.apache.ignite.raft.messages.TestRaftMessagesFactory;
 
 /**
  * Increment and get command.
  */
-public class IncrementAndGetCommand implements WriteCommand {
-    /**
-     * The delta.
-     */
-    private final long delta;
-
-    /**
-     * Constructor.
-     *
-     * @param delta The delta.
-     */
-    public IncrementAndGetCommand(long delta) {
-        this.delta = delta;
-    }
-
+//TODO Move to integration test directory when Maven build is not supported anymore.
+@Transferable(TestMessageGroup.INCREMENT_AND_GET_COMMAND)
+public interface IncrementAndGetCommand extends WriteCommand {
     /**
      * Returns the delta.
      */
-    public long delta() {
-        return delta;
+    long delta();
+
+    static IncrementAndGetCommand incrementAndGetCommand(long delta) {
```

Comment:
Why do you use a static method to create an instance, if there is a factory?
This looks like contradictory patterns to me.

Reply:
Just to be clear, these are test messages. I don't do things like that in production messages.